xarray/core/datatree.py

```diff
@@ -1609,6 +1610,11 @@ def to_netcdf(
         group : str, optional
             Path to the netCDF4 group in the given file to open as the root group
             of the ``DataTree``. Currently, specifying a group is not supported.
+        write_inherited_coords : bool, default: True
```

Comment:
what happens at read-time? Do we normalize and remove duplicated coords by default? If so, I think we should change so that does *not* happen by default.

Reply:
> what happens at read-time? Do we normalize and remove duplicated coords by default?

Yes, this is what happens.

> If so, I think we should change so that does not happen by default.

Can you clarify what you mean here?

Reply:
IIUC the default is to denormalize at write-time and normalize at read-time, where "normalizing" means that coordinates are de-duplicated and inherited from parent groups where possible.

To me, that's confusing in that you really have to opt-in to round-trip to disk exactly what you have in memory.  Also, anyone reading a CF-compliant store with inherited coordinates and writing it out will be surprised when opening it up with another library. FWIW it seems to me that we usually regret this kind of convenience/consistency tradeoff in the long-run.

I don't feel too strongly though. At the very least we should add some docs on how to roundtrip exactly, and how to open and write CF-compliant datasets with coordinate inheritance.

Reply:
The advantage of writing things out this way is that you get the same DataTree from reading out a particular group, i.e.,
```python
tree.to_zarr(path)
child_from_disk = open_datatree(path, group=child)
xarray.testing.assert_equal(tree[child], child_from_disk)
```

I think this will be a little less surprising to users, but overall I agree that it does not matter too much.

Reply:
I'd expect when handling CF compliant data to roundtrip accordingly. 

This might lead to issues on the user side.

Reply:
I've reverted changing this behavior -- the default is now `write_inherited_coords=False`.